atd-vze/src/views/Crashes/CrashChange.js

```diff
@@ -523,19 +502,35 @@ function CrashChange(props) {
   };
 
   /**
-   * Executes graphql queries to update the crash record
+   * Executes graphql query to update the crash record
    * @returns {Promise<void>}
    */
   const executeUpdateCrashRecord = async () => {
-    const updateQueries = await generateUpdateQuery();
-    setSaveStatus("updating crash record");
-    setUpsertRecordQuery(
-      gql`
-        ${updateQueries}
-      `
-    );
-    await sleep(1000);
-    await upsertRecordUpdates();
+    const record = secondaryData["atd_txdot_changes"][0];
+    const recordString = record["record_json"] || "{}";
+    const recordObject =
+      JSON.parse(recordString)[0] || JSON.parse(recordString) || {};
+
+    const crashId = recordObject["crash_id"];
+
+    let changesObject = {};
+
+    selectedFields.forEach(field => {
+      changesObject[field] = recordObject[field];
+    });
+
+    changesObject["updated_by"] = user.email || "DiffView";
+
+    updateCrashRecord({
+      variables: {
+        crashId: crashId,
+        changes: changesObject,
+      },
+    })
+      .then(() => {
+        refetch();
+      })
+      .catch(error => console.error(error));
```

Comment:
was able to simplify this logic a lot and cut down on some of the code and functions that were being used previously